akka-cluster-tools/src/main/resources/reference.conf

```diff
@@ -111,6 +111,13 @@ akka.cluster.client {
   # immediately if the location of the singleton is unknown.
   # Maximum allowed buffer size is 10000.
   buffer-size = 1000
+
+  # If connection to the receiptionist is lost and the client has not been
+  # able to acquire a new connection for this long the client will stop itself.
+  # This makes it possible to watch the cluster client and react on a more permanent
+  # loss of connection with the cluster, for example by accessing some kind of
+  # service registry for an updated set of initial contacts to start a new cluster client with.
+  reconnect-timeout = 1m
```

Comment:
Callbacks are a mess, users will get the threading wrong. If we would need something more than the stop signal it whould be a message. This is actors after all. In my opinion we don't need anything more than the stop signal for the use case presented here.